air/src/stack/stack_flag/mod.rs

```diff
@@ -0,0 +1,969 @@
+        .take(64)
+        .skip(32)
+        .for_each(|v| *v *= not6_5);
+
+    degree_7_intermediate_values
+}
+
+/// This helper function generates all the flag values of degree seven and degree four
+/// stack operations.
+fn set_op_flags_degree_4_and_6<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+) -> ([E; NUM_DEGREE_6_OPS], [E; NUM_DEGREE_4_OPS]) {
+    let mut degree_6_intermediate_values = [E::ZERO; NUM_DEGREE_6_OPS];
+    let mut degree_4_intermediate_values = [E::ZERO; NUM_DEGREE_4_OPS];
+
+    let not_3 = binary_not(frame.op_bit(3));
+    let not_2 = binary_not(frame.op_bit(2));
+    let helper = frame.helper_register();
+
+    degree_4_intermediate_values[0] = not_3 * not_2;
+    degree_4_intermediate_values[1] = frame.op_bit(2) * not_3;
+    degree_4_intermediate_values[2] = not_2 * frame.op_bit(3);
+    degree_4_intermediate_values[3] = frame.op_bit(3) * frame.op_bit(2);
+    degree_4_intermediate_values.copy_within(0..4, 4);
+
+    let not_4 = binary_not(frame.op_bit(4));
+    for i in 0..4 {
+        degree_4_intermediate_values[i] *= not_4;
+        degree_6_intermediate_values[2 * i] = degree_4_intermediate_values[i];
+    }
+    for i in 4..8 {
+        degree_4_intermediate_values[i] *= frame.op_bit(4);
+        degree_6_intermediate_values[2 * i] = degree_4_intermediate_values[i];
+    }
+
+    // helper register is multiplied with all the values to enumerate all the possible
+    // degree 4 operations.
+    degree_4_intermediate_values
+        .iter_mut()
+        .take(8)
+        .for_each(|v| *v *= helper);
+
+    // The degree 6 flag (`10xxxxx`) is clubbed with both 6th bit and it's boolean not value to
+    // reduce the number of multiplications.
+    let not_1 = binary_not(frame.op_bit(1));
+    let degree_six_flag = frame.op_bit(6) * binary_not(frame.op_bit(5));
+    let degree_six_flag_not_1 = degree_six_flag * not_1;
+    let degree_six_flag_yes_1 = degree_six_flag * frame.op_bit(1);
+    for i in (0..16).step_by(2) {
+        degree_6_intermediate_values[i + 1] =
+            degree_6_intermediate_values[i] * degree_six_flag_yes_1;
+        degree_6_intermediate_values[i] *= degree_six_flag_not_1;
+    }
+
+    (degree_6_intermediate_values, degree_4_intermediate_values)
+}
+
+// TESTS
+// ================================================================================================
+
+#[cfg(test)]
+pub mod tests {
+    use crate::stack::stack_flag::OpFlags;
+
+    use super::generate_evaluation_frame;
+    use vm_core::{Felt, Operation, ONE, ZERO};
+
+    #[test]
+    fn op_flags() {
+        for operation in Operation::VALUES.iter() {
+            assert_op_flags(*operation);
+        }
+    }
+
+    // HELPER FUNCTIONS
+    // -------------------------------------------------------------------------------
+
+    /// Asserts operation flags to zero which are not being executed in the current execution frame.
+    fn assert_flags_not_equal(op_flag: OpFlags<Felt>, operation: Operation) {
```

Comment:
I don't think this function works as intended. Specifically, for all operations except the `NOOP` it check only that `NOOP` flag is set to `ZERO` and won't check anything else.

But also, you can probably get rid of this function entirely using the structure I described above.